contracts/secret-storage/TeeManager.sol

```diff
@@ -0,0 +1,600 @@
+        SLASH_MAX_BIPS = _slashMaxBips;
+    }
+
+    //-------------------------------- Overrides start --------------------------------//
+
+    /// @inheritdoc ERC165Upgradeable
+    function supportsInterface(
+        bytes4 interfaceId
+    ) public view virtual override(ERC165Upgradeable, AccessControlUpgradeable) returns (bool) {
+        return super.supportsInterface(interfaceId);
+    }
+
+    /// @inheritdoc UUPSUpgradeable
+    function _authorizeUpgrade(address /*account*/) internal view override onlyRole(DEFAULT_ADMIN_ROLE) {}
+
+    //-------------------------------- Overrides end --------------------------------//
+
+    //-------------------------------- Initializer start --------------------------------//
+
+    /// @notice Thrown when the provided admin address is zero.
+    error TeeManagerZeroAddressAdmin();
+
+    /**
+     * @dev Initializes the contract with the given admin and enclave images.
+     * @param _admin The address of the admin.
+     * @param _images Array of enclave images to initialize.
+     */
+    function initialize(address _admin, EnclaveImage[] memory _images) public initializer {
+        if (_admin == address(0)) revert TeeManagerZeroAddressAdmin();
+
+        __Context_init_unchained();
+        __ERC165_init_unchained();
+        __AccessControl_init_unchained();
+        __UUPSUpgradeable_init_unchained();
+        __AttestationAuther_init_unchained(_images);
+
+        _grantRole(DEFAULT_ADMIN_ROLE, _admin);
+    }
+
+    //-------------------------------- Initializer end --------------------------------//
+
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    IERC20 public immutable STAKING_TOKEN;
+
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    uint256 public immutable MIN_STAKE_AMOUNT;
+
+    /// @notice an integer in the range 0-10^6
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    uint256 public immutable SLASH_PERCENT_IN_BIPS;
+
+    /// @notice expected to be 10^6
+    /// @custom:oz-upgrades-unsafe-allow state-variable-immutable
+    uint256 public immutable SLASH_MAX_BIPS;
+
+    bytes32 public constant JOBS_ROLE = keccak256("JOBS_ROLE");
+
+    Executors public EXECUTORS;
+
+    SecretStore public SECRET_STORE;
+
+    //-------------------------------- TeeManager start ----------------------------------//
+
+    modifier onlySecretStore() {
+        if(_msgSender() != address(SECRET_STORE))
+            revert TeeManagerInvalidSecretStoreManager();
+        _;
+    }
+
+    modifier isValidTeeNodeOwner(address _enclaveAddress) {
+        _isValidTeeNodeOwner(_enclaveAddress);
+        _;
+    }
+
+    function _isValidTeeNodeOwner(address _enclaveAddress) internal view {
+        if (teeNodes[_enclaveAddress].owner != _msgSender())
+            revert TeeManagerInvalidEnclaveOwner();
+    }
+
+    struct TeeNode {
+        uint256 stakeAmount;
+        uint256 reputation;
```

Comment:
we can make reputation part of the executor struct since this parameter is related to job executions only, and not something that is required by secret store and executor both. It's not a common param.